src/sentry/uptime/endpoints/validators.py

```diff
@@ -73,9 +108,13 @@ def validate_mode(self, mode):
             )
 
     def create(self, validated_data):
+        method_headers_body = {
+            k: v for k, v in validated_data.items() if k in {"method", "headers", "body"}
+        }
```

Comment:
Why do it like this vs just explicitly having

```py
method=validated_data["methods"],
headers=validated_data["headers"],
body=validated_data["body"],
```

Reply:
Since these are optional, this filters out things that weren't passed so that the defaults from the method we call are set.

Although maybe it'd be better to set a default on the fields and just always pass it?

Reply:
Ah yeah makes sense. I am +1 to defaults

Reply:
I'll handle this as part of the update api. Adding defaults might have some unexpected side effects and I think it's better to get this merged

Reply:
👍